pkg/controller/propagator/propagation.go

```diff
@@ -310,6 +324,16 @@ func (r *ReconcilePolicy) handleDecision(instance *policiesv1.Policy, decision a
 			// Make sure the Owner Reference is cleared
 			replicatedPlc.SetOwnerReferences(nil)
 
```

Comment:
Since you previously mentioned that you were concerned about performance, you could consider each call to `r.handleDecision(instance, decision)` to be in its own go routine with an upper limit. In the case where a single policy goes to a thousand managed clusters, this would speed things up quite a bit.

Reply:
As this is not something introduced in this issue or something caused due to the addition of templates support. I dont think It should handled in this PR. Please open a separate issue to tracj/fix this.